### DISCUSSION TITLE
Is it insecure to include my login page in nuxt application?

### DISCUSSION THREAD

**Discussion by arai03** [nuxt/nuxt#9052] [Questions (Nuxt 2)] [answered]
if you include your login page in nuxt project, then the user is receiving all of your code before they're even authenticated.
Is there a way to solve this?

**Comment by danielroe** [comment, marked as answer]
@arai03 As a matter of best practice, you should assume that everything in your webpack bundle is public. Secrets or sensitive URLs should not be embedded in your webpack code. The right approach, however, depends a lot on your threat model and use cases. An option to consider is using a serverless API (or server endpoint) to guard access to your secrets (for example, to proxy access to a payment provider or an email service).
